corehq/apps/users/models_role.py

```diff
@@ -118,15 +115,23 @@ def create(cls, domain, name, permissions=None, assignable_by=None, **kwargs):
         return role
 
     @classmethod
-    @quickcache(['domain'])
     def commcare_user_default(cls, domain):
-        try:
-            return UserRole.objects.get_commcare_user_default(domain)
-        except UserRole.DoesNotExist:
-            notify_error("Domain is missing default commcare user role", {
+        """This will get the default mobile worker role for the domain. If one does not exist it
+        will create a new role.
+
+        Note: the role should exist for all domains but errors during domain registration can leave
+        domains improperly configured."""
+        from corehq.apps.users.role_utils import UserRolePresets
+        role, created = UserRole.objects.get_or_create(domain=domain, is_commcare_user_default=True, defaults={
+            "name": UserRolePresets.MOBILE_WORKER
+        })
+        if created:
+            notify_error("Domain was missing default commcare user role", {
                 "domain": domain
             })
-        return None
+            permissions = UserRolePresets.INITIAL_ROLES[UserRolePresets.MOBILE_WORKER]()
+            role.set_permissions(permissions.to_list())
+        return role
```

Comment:
Nice, I really like this commit - feels like a good compromise solution to the circumstances.